src/Backend/Modules/MediaLibrary/Domain/MediaFolder/MediaFolderDataTransferObject.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Backend\Modules\MediaLibrary\Domain\MediaFolder;
+
+class MediaFolderDataTransferObject
+{
+    /** @var MediaFolder */
+    protected $mediaFolderEntity;
+
+    /** @var MediaFolder|null */
+    public $parent;
+
+    /** @var string */
+    public $name;
+
+    /** @var int */
+    public $userId;
+
+    /**
+     * CreateMediaFolder constructor.
+     *
+     * @param MediaFolder|null $mediaFolder
+     */
+    public function __construct(MediaFolder $mediaFolder = null)
+    {
+        $this->mediaFolderEntity = $mediaFolder;
+
+        if ($this->hasExistingMediaFolder()) {
```

Comment:
I would write this as if not return, that way you have less code nested inside an if